src/material/datepicker/calendar-range-selection-strategy.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, InjectionToken} from '@angular/core';
+import {DateAdapter} from '@angular/material/core';
+import {DateRange} from './date-selection-model';
+
+/** Injection token used to customize the date range selection behavior. */
+export const MAT_CALENDAR_RANGE_SELECTION_STRATEGY =
+    new InjectionToken<MatCalendarRangeSelectionStrategy<any>>(
+        'MAT_CALENDAR_RANGE_SELECTION_STRATEGY');
+
+/** Object that can be provided in order to customize the date range selection behavior. */
+export interface MatCalendarRangeSelectionStrategy<D> {
+  /** Called when the user has finished selecting a value. */
+  selectionFinished(date: D | null, currentRange: DateRange<D>, event: Event): DateRange<D>;
```

Comment:
Hmm, good point. I was looking for something that conveys that this will be the final event in the sequence. The other option I had in mind was `selectionDone`, but it's not much better. We could call it `pointerUp`, but that's misleading since it could be the result of a keyboard event.

Reply:
I see so basically it'll be something like this after we add touch support:
- selectionStarted = mousedown/keydown/dragenter
- selectionFinished = mouseup/keyup/drop

I guess I can't really think of better names, I think we just need to make sure its documented well and explains what type of events go to what method

Reply:
Yeah exactly, with potentially a `selectionMoved` to update the preview. I'll leave the name as it is for now and address the rest of the comments.